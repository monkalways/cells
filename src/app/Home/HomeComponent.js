import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid, withStyles } from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

import Header from '../common/Header';
import constants from '../constants';
import logo from '../images/EPSlogo.png';
import GotoCell from './GotoCell';
import GotoActivityRoom from './GotoActivityRoom';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  versions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
  })).isRequired,
  getVersions: PropTypes.func.isRequired,
};

class HomeComponent extends Component {
  componentDidMount() {
    const { getVersions } = this.props;
    getVersions();
  }

  render() {
    const { classes, versions } = this.props;
    return (
      <Grid container justify="center">
        <Grid container className={classes.container} justify="center">
          <Grid item xs={12}>
            <Header versions={versions} />
          </Grid>
          <Grid item>
            <img
              src={logo}
              alt="EPS Logo"
              width={300}
              className={classes.logo}
            />
          </Grid>
          <Grid item container xs={12}>
            <Grid item xs={12} sm={6}>
              <GotoCell />
            </Grid>
            <Grid item xs={12} sm={6}>
              <GotoActivityRoom />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

HomeComponent.propTypes = propTypes;

export default compose(
  withStyles((theme) => ({
    container: {
      height: constants.LAYOUT.height,
      background: theme.palette.background.default,
      width: constants.LAYOUT.width,
    },
    item: {
      marginLeft: theme.spacing.unit * 2,
      marginRight: theme.spacing.unit * 2,
    },
    logo: {
      marginTop: theme.spacing.unit * 6,
    },
  })),
  withRouter,
)(HomeComponent);
