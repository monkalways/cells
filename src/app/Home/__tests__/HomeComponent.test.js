import React from 'react';
import { shallow } from 'enzyme';

import { HomeComponent } from '../HomeComponent';

describe('HomeComponent', () => {
  let classes;
  let push;
  let versions;
  let getVersions;

  beforeEach(() => {
    classes = {
      paper: 'paper',
      form: 'form',
      submit: 'submit',
    };
    push = jest.fn();
    versions = [
      {
        name: 'Cell App',
        version: '1.0.0',
      },
      {
        name: 'Detainee Management Service',
        version: '1.0.0',
      },
      {
        name: 'EPROS Service',
        version: '1.0.0',
      },
    ];
    getVersions = jest.fn();
  });

  const setup = () => shallow(<HomeComponent
    classes={classes}
    history={{ push }}
    versions={versions}
    getVersions={getVersions}
  />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call get versions when componentDidMount is called', () => {
    const wrapper = setup();
    wrapper.instance().componentDidMount();

    expect(getVersions).toBeCalled();
  });
});
