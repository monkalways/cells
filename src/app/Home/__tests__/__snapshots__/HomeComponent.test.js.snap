// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render 1`] = `
<WithStyles(Grid)
  container={true}
  justify="center"
>
  <WithStyles(Grid)
    container={true}
    justify="center"
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(HeaderComponent)
        versions={
          Array [
            Object {
              "name": "Cell App",
              "version": "1.0.0",
            },
            Object {
              "name": "Detainee Management Service",
              "version": "1.0.0",
            },
            Object {
              "name": "EPROS Service",
              "version": "1.0.0",
            },
          ]
        }
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <img
        alt="EPS Logo"
        src="EPSlogo.png"
        width={300}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(Grid)
        item={true}
        sm={6}
        xs={12}
      >
        <WithStyles(withRouter(GotoCellComponent)) />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm={6}
        xs={12}
      >
        <WithStyles(withRouter(GotoActivityRoomComponent)) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
